src/start_env/mod.rs

```diff
@@ -86,67 +86,97 @@ struct Services {
         .iter()
         .map(|hd_key| {
             hd_key
-                .derive_priv(
-                    &Secp256k1::new(),
-                    &vec![
-                        ChildNumber::from_hardened_idx(44).unwrap(),
-                        ChildNumber::from_hardened_idx(1).unwrap(),
-                        ChildNumber::from_hardened_idx(0).unwrap(),
-                        ChildNumber::from_normal_idx(0).unwrap(),
-                        ChildNumber::from_normal_idx(0).unwrap(),
-                    ],
-                )
-                .unwrap()
-                .private_key
-                .key
+                .derive_priv(&Secp256k1::new(), &derivation_path)
+                .map(|secret_key| secret_key.private_key.key)
         })
-        .collect::<Vec<_>>();
-
-    let env_file_path = temp_fs::env_file_path();
-    temp_fs::create_env_file()
-        .unwrap_or_else(|_| panic!("Could not create {:?} file", env_file_path));
-
-    let docker_network_create = create_network();
-
+        .collect::<Result<Vec<_>, _>>()?;
+    let env_file_path = temp_fs::env_file_path()?;
+    let docker_network_create = create_network().map_err(|e| {
+        eprintln!("Issue creating Docker network: {:?}", e);
+    });
     let bitcoin_node = start_bitcoin_node(&env_file_path, bitcoin_priv_keys).map_err(|e| {
         eprintln!("Issue starting Bitcoin node: {:?}", e);
     });
-
     let ethereum_node =
         start_ethereum_node(&env_file_path, ethereum_priv_keys.clone()).map_err(|e| {
             eprintln!("Issue starting Ethereum node: {:?}", e);
         });
+    let btsieves = {
+        let (path, string) = temp_fs::temp_folder()?;
+        start_btsieves(&env_file_path, path, string).map_err(|e| {
+            eprintln!("Issue starting btsieves: {:?}", e);
+        })
+    };
+    let cnds = {
+        let (path, string) = temp_fs::temp_folder()?;
+        start_cnds(&env_file_path, path, string).map_err(|e| {
+            eprintln!("Issue starting cnds: {:?}", e);
+        })
+    };
+    Ok((
+        bitcoin_hd_keys,
+        ethereum_priv_keys,
+        env_file_path,
+        docker_network_create,
+        bitcoin_node,
+        ethereum_node,
+        btsieves,
+        cnds,
+    ))
+}
 
-    let btsieves = start_btsieves(&env_file_path).map_err(|e| {
-        eprintln!("Issue starting btsieves: {:?}", e);
-    });
-
-    let cnds = start_cnds(&env_file_path).map_err(|e| {
-        eprintln!("Issue starting cnds: {:?}", e);
-    });
+fn start_all(runtime: &mut Runtime) -> Result<Services, Error> {
+    let (
+        bitcoin_hd_keys,
+        ethereum_priv_keys,
+        env_file_path,
+        docker_network_create,
+        bitcoin_node,
+        ethereum_node,
+        btsieves,
+        cnds,
+    ) = all_futures()?;
 
-    let mut runtime = Runtime::new().expect("Could not get runtime");
+    temp_fs::create_env_file()?;
+    let env_file_str = temp_fs::env_file_str()?;
```

Comment:
:wrench: Would much rather see this returned as part of `create_env_file`, especially since it's only used for logging errors.